py/redrock/external/desi.py

```diff
@@ -580,6 +592,9 @@ def rrdesi(options=None, comm=None):
 
     parser.add_argument("--nminima", type=int, default=3,
         required=False, help="the number of redshift minima to search")
+    
+    parser.add_argument("--prior_sigma", type=float, default=None,
+        required=False, help="sigma to add as prior in solving linear equation, 1/sig**2 will be added")
```

Comment:
The command line options are getting confusingly similar, e.g. `--prior_sigma` has nothing to do with `--priors`.  I suggest that we standardize all archetype-related options to start with `--archetype`, e.g.
```
-deg_legendre -> --archetype-legendre-degree
--per-camera  -> --archetype-legendre-percamera
--prior_sigma -> --archetype-legendre-prior
-n_nearest    -> --archetype-nnearest (?)
```
We might also want to consider a single `--archetype-legendre` that turns on the other 3 legendre-related terms with the recommended defaults.

Let's also rename `-nz` (number of finer redshift to be used around best fit redshifts) to `--zminfit_npoints` or something similar to try to make the option name clearer that it is number of points fit around each minimum, not the total number of redshifts to fit.

Reply:
I have cleaned up the arguments now. I have added a flag `--archetype-legendre` that will run archetypes with default values. Otherwise user can also provide desired arguments.